hermes_core/util/cdf.py

```diff
@@ -0,0 +1,612 @@
+            raise KeyError(f"CDFWriter does not contain data variable {name}")
+        # Get the Data and Attrs for the named variable
+        var_data = self.variable_data[name]
+        var_attrs = self.variable_attrs[name] if name in self.variable_attrs else {}
+        return (var_data, var_attrs)
+
+    def __setitem__(self, name, data):
+        """
+        Function to set a data variable conained in the CDFWriter class.
+        """
+        # Set the Data for the named variable
+        self.variable_data[name] = data
+        # Set the Attrs for the named variable
+        if name not in self.variable_attrs:
+            self.variable_attrs[name] = {}
+
+    def __contains__(self, name):
+        """
+        Function to see whether a data variable is in the CDFWriter class.
+        """
+        return name in self.variable_data
+
+    def __delitem__(self, name):
+        """
+        Function to remove a data variable and attrs from the CDFWriter class.
+        """
+        # Remove from variable data
+        self.variable_data.pop(name, None)
+        # Remove from variable attributes
+        self.variable_attrs.pop(name, None)
+
+    def __iter__(self):
+        """
+        Function to iterate over data variables and attributes in the CDFWriter class.
+        """
+        for var_name in self.variable_data.keys():
+            var_data = self.variable_data[var_name]
+            var_attrs = self.variable_attrs[var_name]
+            yield (var_name, var_data, var_attrs)
+
+    def _load_default_schema_data(self):
+        # The Default Schema file is contained in the `hermes_core/data` directory
+        default_schema_path = str(Path(hermes_core.__file__).parent / "data" / DEFAULT_SCHEMA_FILE)
+        # Load the Schema
+        self.load_schema_data(schema_data_path=default_schema_path)
+
+    def _load_default_attributes(self):
+        # The Default Attributes file is contained in the `hermes_core/data` directory
+        default_attributes_path = str(
+            Path(hermes_core.__file__).parent / "data" / DEFAULT_ATTRIBUTES_FILE
+        )
+        # Load the Schema
+        self.add_attributes_from_yaml(attributes_path=default_attributes_path)
+
+    def load_schema_data(self, schema_data_path):
+        """
+        Function to load schema for CDF formatting data from a Yaml file.
+
+        Parameters
+        ----------
+        schema_data_file: `str`
+            Path to schem file to be used for CDF formatting.
+
+        """
+        assert isinstance(schema_data_path, str)
+        assert Path(schema_data_path).exists()
+        # Load the Yaml file to Dict
+        schema_data = {}
+        with open(schema_data_path, "r") as f:
+            try:
+                schema_data = yaml.safe_load(f)
+            except yaml.YAMLError as exc:
+                log.critical(exc)
+        # Set the Current Schema
+        self.schema = schema_data
+
+    def add_attributes_from_list(self, attributes: list):
```

Comment:
Aren't the attributes just a dictionary? There already exists a way to add a dict to another. Also shouldn't the input be a dictionary as well, key and value pairs for the attributes?

Reply:
Yeah, we can have whatever kinds of functions to add global attributes. I'll add one that takes in a dict as you suggest but they're all easy to write.